tests/test_aggregates.py

```diff
@@ -0,0 +1,319 @@
+import pytest
+
+from functools import partial
+from tempfile import TemporaryDirectory
+
+import signac
+from flow.aggregates import aggregator, get_aggregate_id
+
+
+@pytest.fixture
+def list_of_aggregates():
+    def helper_default_aggregator_function(jobs):
+        return [jobs]
+
+    def helper_non_default_aggregator_function(jobs):
+        for job in jobs:
+            yield (job,)
+
+    # The below list contains 9 distinct aggregator objects and some duplicates.
+    return [
+        aggregator(), aggregator(), aggregator(helper_default_aggregator_function),
+        aggregator(helper_non_default_aggregator_function),
+        aggregator(helper_non_default_aggregator_function),
+        aggregator.groupsof(1), aggregator.groupsof(1),
+        aggregator.groupsof(2), aggregator.groupsof(3), aggregator.groupsof(4),
+        aggregator.groupby('even'), aggregator.groupby('even'),
+        aggregator.groupby('half', -1), aggregator.groupby('half', -1),
+        aggregator.groupby(['half', 'even'], default=[-1, -1])
+    ]
+
+
+class AggregateProjectSetup:
+    project_class = signac.Project
+    entrypoint = dict(path='')
+
+    @pytest.fixture
+    def setUp(self, request):
+        self._tmp_dir = TemporaryDirectory(prefix='flow-aggregate_')
+        request.addfinalizer(self._tmp_dir.cleanup)
+        self.project = self.project_class.init_project(
+            name='AggregateTestProject',
+            root=self._tmp_dir.name)
+
+    def mock_project(self):
+        project = self.project_class.get_project(root=self._tmp_dir.name)
+        for i in range(10):
+            even = i % 2 == 0
```

Comment:
```suggestion
            even = (i % 2) == 0
```

Relying on operator precedence when using anything other than pure arithmetic operators always makes me uneasy.